lib/Dialect/ModArith/IR/ModArithOps.td

```diff
@@ -20,68 +20,76 @@ class ModArith_Op<string mnemonic, list<Trait> traits = [Pure]> :
 }
 
 // type conversion operations
-def ModArith_EncapsulateOp : ModArith_Op<"encapsulate", [Pure, ElementwiseMappable]> {
+def ModArith_EncapsulateOp : ModArith_Op<"encapsulate", [Pure]> {
   let summary = "encapsulate an integer into a mod_arith type";
 
   let description = [{
-    `mod_arith.encapsulate` converts the integer to be of mod_arith type.
+    `mod_arith.encapsulate` converts the integer to be of mod_arith or RNS type.
 
     Examples:
     ```
     mod_arith.encapsulate %c0 : i32 -> mod_arith.int<65537 : i32>
-    mod_arith.encapsulate %c1 : i64 -> mod_arith.int<65537>
+    mod_arith.encapsulate %c1 : i64 -> mod_arith.int<65537 : i64>
+    mod_arith.encapsulate %c2 : tensor<4x2xi32> -> tensor<4xrns.rns<mod_arith.int<5 : i32>, mod_arith.int<17 : i32>>>
     ```
   }];
 
   let arguments = (ins
     SignlessIntegerLike:$input
   );
-  let results = (outs ModArithLike:$output);
+  let results = (outs ModArithOrRNSLike:$output);
+  let hasVerifier = 1;
   let assemblyFormat = "operands attr-dict `:` type($input) `->` type($output)";
 }
 
-def ModArith_ExtractOp : ModArith_Op<"extract", [Pure, ElementwiseMappable]> {
```

Comment:
Nit: is there a reason to remove ElementwiseMappable here, since the type constraints still allow tensor types?